src/services/user/autoFollowUserAnon.js

```diff
@@ -17,37 +17,39 @@ const autoFollowUserAnon = async (req, res) => {
       userFollowUserList.map(async (followUser) => {
         const anonId = await UsersFunction.findAnonymousUserId(followUser.user_id_followed);
 
-        const [followAnonUser, isUnfollowByUser] = await Promise.all([
-          UserFollowUser.findOne({
-            where: {
+        if (anonId) {
+          const [followAnonUser, isUnfollowByUser] = await Promise.all([
+            UserFollowUser.findOne({
+              where: {
+                user_id_follower: followUser.user_id_follower,
+                user_id_followed: anonId,
+                is_anonymous: true
+              }
+            }),
+            UserFollowUserHistory.findOne({
+              where: {
+                user_id_follower: followUser.user_id_follower,
+                user_id_followed: anonId,
+                action: 'out'
+              }
+            })
+          ]);
+
+          if (!followAnonUser && !isUnfollowByUser) {
+            userFollowerAnonData.push({
+              follow_action_id: uuidv4(),
               user_id_follower: followUser.user_id_follower,
               user_id_followed: anonId,
               is_anonymous: true
-            }
-          }),
-          UserFollowUserHistory.findOne({
-            where: {
+            });
+
+            userFollowerAnonDataHistory.push({
               user_id_follower: followUser.user_id_follower,
               user_id_followed: anonId,
-              action: 'out'
-            }
-          })
-        ]);
-
-        if (!followAnonUser && !isUnfollowByUser) {
-          userFollowerAnonData.push({
-            follow_action_id: uuidv4(),
-            user_id_follower: followUser.user_id_follower,
-            user_id_followed: anonId,
-            is_anonymous: true
-          });
-
-          userFollowerAnonDataHistory.push({
-            user_id_follower: followUser.user_id_follower,
-            user_id_followed: anonId,
-            action: 'in',
-            source: `script`
-          });
+              action: 'in',
+              source: `script`
+            });
+          }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic and error handling improvements look good. However, there's a potential performance issue here. You're using `Array.prototype.map` which returns a new array, but you're not using the returned array. If you just want to execute side effects for each element, consider using `Array.prototype.forEach` instead. Also, you're performing async operations inside the loop, which means these operations will run concurrently. This could lead to high memory usage if `userFollowUserList` is large. Consider using `for...of` loop with async/await to handle promises one by one.

```diff
- userFollowUserList.map(async (followUser) => {
+ for (const followUser of userFollowUserList) {
    const anonId = await UsersFunction.findAnonymousUserId(followUser.user_id_followed);

    if (anonId) {
      const [followAnonUser, isUnfollowByUser] = await Promise.all([
        UserFollowUser.findOne({
          where: {
            user_id_follower: followUser.user_id_follower,
            user_id_followed: anonId,
            is_anonymous: true
          }
        }),
        UserFollowUserHistory.findOne({
          where: {
            user_id_follower: followUser.user_id_follower,
            user_id_followed: anonId,
            action: 'out'
          }
        })
      ]);

      if (!followAnonUser && !isUnfollowByUser) {
        userFollowerAnonData.push({
          follow_action_id: uuidv4(),
          user_id_follower: followUser.user_id_follower,
          user_id_followed: anonId,
          is_anonymous: true
        });

        userFollowerAnonDataHistory.push({
          user_id_follower: followUser.user_id_follower,
          user_id_followed: anonId,
          action: 'in',
          source: `script`
        });
      }
    }
- })
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->